services/src/main/java/za/co/sfy/lucid/sams/rest/mapper/SubjectCriteriaMapper.java

```diff
@@ -0,0 +1,202 @@
+package za.co.sfy.lucid.sams.rest.mapper;
+
+import java.sql.Date;
+import org.springframework.stereotype.Component;
+import za.co.sfy.lucid.sams.rest.vo.data.writer.SubjectCriteriaRequest;
+import za.co.sfy.sams.lucid.schema.SubjectCriteria;
+
+/**
+ * @author Azeem
+ */
+@Component
+public class SubjectCriteriaMapper {
+
+    public SubjectCriteria SubjectCriteriaRequestToSubjectCriteria(SubjectCriteriaRequest subjectcriteriaRequest) {
+
+        SubjectCriteria subjectcriteria = new SubjectCriteria();
+
+     
+
+        Integer subjectID = subjectcriteriaRequest.getSubjectID();
+        if (null != subjectID) {
+            subjectcriteria.setSubjectID(subjectcriteriaRequest.getSubjectID());
+        }
+
+        Integer criterionID  = subjectcriteriaRequest.getCriterionID();
+        if (null != criterionID) {
+            subjectcriteria.setCriterionID(subjectcriteriaRequest.getCriterionID());
+        }
+
+        Short description = subjectcriteriaRequest.getDescription();
+        if (null != description) {
+               subjectcriteria.setDescription(subjectcriteriaRequest.getDescription());
+        }
+
+         Integer grade = subjectcriteriaRequest.getGrade();  
+        if (null != grade) {
+               subjectcriteria.setGrade(subjectcriteriaRequest.getGrade());
+        }
+
+        Integer weighting = subjectcriteriaRequest.getWeighting();  
+        if (null != weighting) {
+               subjectcriteria.setGrade(subjectcriteriaRequest.getWeighting());
+        }
+
+          Short subjectlevel = subjectcriteriaRequest.getSubjectLevel();
+        if (null != subjectlevel) {
+               subjectcriteria.setSubjectLevel(subjectcriteriaRequest.getSubjectLevel());
+        }
+
+
+           Integer criterionscore = subjectcriteriaRequest.getCriterionScore();
+        if (null != criterionscore) {
+            subjectcriteria.setCriterionScore(subjectcriteriaRequest.getCriterionScore());
+        }
+
+         Short datayear = subjectcriteriaRequest.getDataYear();
+        if (null != datayear) {
+               subjectcriteria.setSubjectLevel(subjectcriteriaRequest.getDataYear());
+        }
```

Comment:
you not making use of the variables you initialized, please fix this throughout your changes.

e.g.  subjectcriteria.setSubjectLevel(subjectcriteriaRequest.getDataYear()); should be  subjectcriteria.setSubjectLevel(datayear);